common/osqp_interface/CMakeLists.txt

```diff
@@ -19,14 +19,16 @@ project(osqp_interface)
 # require that dependencies from package.xml be available
 find_package(ament_cmake_auto REQUIRED)
 find_package(Eigen3 REQUIRED)
-find_package(osqp REQUIRED)
-get_target_property(OSQP_INCLUDE_DIR osqp::osqp INTERFACE_INCLUDE_DIRECTORIES)
 
 ament_auto_find_build_dependencies(REQUIRED
   ${${PROJECT_NAME}_BUILD_DEPENDS}
   ${${PROJECT_NAME}_BUILDTOOL_DEPENDS}
 )
 
+find_package(osqp REQUIRED)
+get_target_property(OSQP_INCLUDE_SUB_DIR osqp::osqp INTERFACE_INCLUDE_DIRECTORIES)
+get_filename_component(OSQP_INCLUDE_DIR ${OSQP_INCLUDE_SUB_DIR} PATH)
```

Comment:
Could you explain this a bit more in detail? :pray: 
If it should be after `ament_auto_find_build_dependencies`, it's good to add some comments.

Reply:
I guess we should find `osqp` after finding `osqp_vendor` in `ament_auto_find_build_dependencies` .

Reply:
I see, then could you add the comment? :pray:

Reply:
I added comment at https://github.com/autowarefoundation/autoware.universe/pull/776/commits/063fc4247ffce699198c98e0f995b055f4d6a340.